tasks/eth/base-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,293 @@
+    }
+
+    function getCodeExceptions() internal pure override returns (address[] memory) {
+        address[] memory shouldHaveCodeExceptions = new address[](4);
+
+        shouldHaveCodeExceptions[0] = systemConfigOwner;
+        shouldHaveCodeExceptions[1] = batchSenderAddress;
+        shouldHaveCodeExceptions[2] = p2pSequencerAddress;
+        shouldHaveCodeExceptions[3] = batchInboxAddress;
+
+        return shouldHaveCodeExceptions;
+    }
+
+    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/configs/mainnet/base.toml
+    function _getContractSet() internal view returns (Types.ContractSet memory _proxies) {
+        string memory addressesToml;
+
+        // Read addresses json
+        try vm.readFile(
+            string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/configs/mainnet/base.toml")
+        ) returns (string memory data) {
+            addressesToml = data;
+        } catch {
+            revert("Failed to read lib/superchain-registry/superchain/configs/mainnet/base.toml");
+        }
+
+        _proxies.L1CrossDomainMessenger = stdToml.readAddress(addressesToml, "$.addresses.L1CrossDomainMessengerProxy");
+        _proxies.L1StandardBridge = stdToml.readAddress(addressesToml, "$.addresses.L1StandardBridgeProxy");
+        _proxies.OptimismMintableERC20Factory =
+            stdToml.readAddress(addressesToml, "$.addresses.OptimismMintableERC20FactoryProxy");
+        _proxies.OptimismPortal = stdToml.readAddress(addressesToml, "$.addresses.OptimismPortalProxy");
+        _proxies.OptimismPortal2 = stdToml.readAddress(addressesToml, "$.addresses.OptimismPortalProxy");
+        _proxies.SystemConfig = stdToml.readAddress(addressesToml, "$.addresses.SystemConfigProxy");
+        _proxies.L1ERC721Bridge = stdToml.readAddress(addressesToml, "$.addresses.L1ERC721BridgeProxy");
+        _proxies.DisputeGameFactory = dgfProxy;
+
+        // Read superchain.toml
+        string memory chainConfig;
+        string memory path =
+            string.concat("/lib/superchain-registry/superchain/configs/mainnet/superchain.toml");
+        try vm.readFile(string.concat(vm.projectRoot(), path)) returns (string memory data) {
+            chainConfig = data;
+        } catch {
+            revert(string.concat("Failed to read ", path));
+        }
+
+        _proxies.ProtocolVersions = stdToml.readAddress(chainConfig, "$.protocol_versions_addr");
+        _proxies.SuperchainConfig = stdToml.readAddress(chainConfig, "$.superchain_config_addr");
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](6);
+        allowed[0] = address(proxies.OptimismPortal2);
+        allowed[1] = address(proxies.SystemConfig);
+        allowed[2] = address(ownerSafe);
+        allowed[3] = address(securityCouncilSafe);
+        allowed[4] = address(fndSafe);
+        // allowed[4] = expectedLivenessGuard;
```

Comment:
This comment seems forgotten as the livenessGuard should not be modified during the upgrade.